test/e2e/manifest_test.go

```diff
@@ -369,7 +369,25 @@ var _ = Describe("Podman manifest", func() {
 		session.WaitWithDefaultTimeout()
 		Expect(session).To(ExitWithError())
 
-		session = podmanTest.Podman([]string{"manifest", "rm", "foo1", "foo2"})
+		// push --rm after pull image (#15033)
+		session = podmanTest.Podman([]string{"pull", "quay.io/libpod/testdigest_v2s2"})
+		session.WaitWithDefaultTimeout()
+		Expect(session).Should(Exit(0))
+
+		session = podmanTest.Podman([]string{"manifest", "create", "bar"})
+		session.WaitWithDefaultTimeout()
+		Expect(session).Should(Exit(0))
+		session = podmanTest.Podman([]string{"manifest", "add", "bar", "quay.io/libpod/testdigest_v2s2"})
+		session.WaitWithDefaultTimeout()
+		Expect(session).Should(Exit(0))
+		session = podmanTest.Podman([]string{"manifest", "push", "--rm", "bar", "dir:" + dest})
+		session.WaitWithDefaultTimeout()
+		Expect(session).Should(Exit(0))
+		session = podmanTest.Podman([]string{"manifest", "inspect", "bar"})
+		session.WaitWithDefaultTimeout()
+		Expect(session).To(ExitWithError())
```

Comment:
Can we also check the output of the error?

Reply:
Thank you for reviewing.

This session is expected to output about `short-name resolved` error.

```
# podman images -q foo

# podman manifest inspect foo
Error: short-name "foo" did not resolve to an alias and no unqualified-search registries are defined in "/etc/containers/registries.conf"
```

However, I think it is enough to check the manifest list `bar` has been removed.
Therefore, I replace `manifest inspect bar` to `images -q bar`.